### ISSUE TITLE
[dev] Ignore cached flag when hashing outpoint

### ISSUE THREAD

**Issue by m-kus** [keep-starknet-strange/raito#233]
**Context**

Currently `TxOut` contains `cached` field which is a hint that tells the client this output will be spent within the same block it is created. Since `OutPoint` includes `TxOut` it contributes to the outpoint hash, which becomes a leaf in the Utreexo accumulator. However, the Utreexo node cannot provide this hint (easily), and generally speaking it is not part of the data (it is a hint).

**Task**

- Add custom implementation of the `Hash` trait for `TxOut` to exclude `cached` field
- Update data generation scripts accordingly
- Regenerate integration tests for utreexo

Depends on https://github.com/keep-starknet-strange/raito/pull/229 (should be merged first)

**Comment by mubarak23** [comment]
T

**Comment by PavitraAgarwal21** [comment]
@m-kus Can I take this

**Comment by bloomingpeach** [comment]
May I work on this @m-kus?

**Comment by baitcode** [comment]
```
#[test]
    pub fn test_outpoint_poseidon_hash() {
        let mut test_outpoint = OutPoint {
            txid: DigestTrait::new([1, 2, 3, 4, 5, 6, 7, 8]),
            vout: 2,
            // https://learnmeabitcoin.com/explorer/tx/0437cd7f8525ceed2324359c2d0ba26006d92d856a9c20fa0241106ee5a597c9#output-0
            data: TxOut {
                value: 50_u64,
                pk_script: @"410411db93e1dcdb8a016b49840f8c53bc1eb68a382e97b1482ecad7b148a6909a5cb2e0eaddfb84ccf9744464f82e160bfa9b8b64f9d4c03f999b8643f656b412a3ac",
                cached: false,
            },
            block_height: 9,
            median_time_past: 1650000000,
            is_coinbase: false,
        };
        assert_eq!(
            test_outpoint.hash(),
            3426256427357770988835517595549266311441229240020614569376880225204214993534
        );
    }
    ```

shall I change the hash here, or leave cached flag in the hasheing but with always false value?
I'm not sure what are the policies?

**Comment by m-kus** [comment]
@baitcode the hashes in unit tests will change once you remove the cached field, if that's what you mean
